src/omni/client.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import annotations
+
+from typing import Literal
+
+import requests
+
+from .env import OmniEnv
+
+
+class OmniApiClient:
+    """Class for interacting with the Omni REST API. There are low level functions for making direct requests to the
+    API (get, post, put, delete). These methods take a "path" arg that is equivalent to the path given in the Omni
+    API docs. Additionally the class includes convenience methods for common tasks.
+    """
+
+    def __init__(
+        self, organization_name: str | None = None, api_token: str | None = None
+    ) -> None:
+        _organization_name = organization_name or OmniEnv.ORGANIZATION_NAME
+        self.base_url = f"https://{_organization_name}.omniapp.co/api"
+        self.api_token = api_token or OmniEnv.API_KEY
+        if not _organization_name:
```

Comment:
Given the pattern I just added in my open PR, we should discuss a different approach where the base url logic is shared between this API client class and the Dashboard embedder

Reply:
I was thinking about making that OmniEnv class take the arguments we pass into the constructors and then it would be responsible for using those args or falling back to the env vars. Then we would use OmniEnv.organization_name. I could put that together today pretty quickly. I actually poked around looking for a python library that handled this and was surprised to not find anything. I suspect I was searching for the right terms though